nat-lab/tests/helpers.py

```diff
@@ -421,3 +423,38 @@ async def ping_between_all_nodes(env: Environment) -> None:
         )
         if not client.is_node(node)
     ])
+
+
+async def send_https_request(
+    connection, endpoint, method, ca_cert_path, data=None, expect_response=True
+):
+    curl_command = [
+        "curl",
+        "--cacert",
+        ca_cert_path,
+        "-X",
+        method,
+        endpoint,
+        "-H",
+        "Content-Type: application/json",
+    ]
+
+    if data:
+        curl_command.extend(["-d", data])
+
+    process = await connection.create_process(curl_command).execute()
+    response = process.get_stdout()
+    if expect_response:
+        try:
+            return json.loads(response)
+        except json.JSONDecodeError:
+            assert False, f"Expected JSON response but got: {response}"
+    return None
```

Comment:
Why not use requests ([example](https://requests.readthedocs.io/en/stable/user/advanced/#ssl-cert-verification))? It's already in our pipfile